js/client-chat.js

```diff
@@ -1455,15 +1455,17 @@
 			var lastMessageDates = Tools.prefs('logtimes') || (Tools.prefs('logtimes', {}), Tools.prefs('logtimes'));
 			if (!lastMessageDates[Config.server.id]) lastMessageDates[Config.server.id] = {};
 			var lastMessageDate = lastMessageDates[Config.server.id][this.id] || 0;
-			var mayNotify = msgTime > lastMessageDate && userid !== app.user.get('userid');
+			// because the time offset to the server can vary slightly, subtract it to not have it affect comparisons between dates
+			var serverMsgTime = msgTime - ~~(this.timeOffset);
```

Comment:
Oh, `~~` seems to round closer-to-0, cast to Number, and turn `NaN` into 0.

Sometimes things like this slip through, but my personal preference is to use `(this.timeOffset || 0)` if the worry is that it might be `undefined`.